syntax = "proto2";

package island;

import "constant.proto";

option optimize_for = LITE_RUNTIME;

option java_package = "com.box.island.proto";  
option java_outer_classname = "Struct";  

message GameServer{
	required int32 id = 1;
    required string ip = 2;
    required int32 port = 3;
}


message Player {
    required int32 uid = 1;
    required int32 team = 2;
    required int32 color = 3;
    required string name = 4;  
    required ForceType force_type = 5;
    required bool is_online = 6;
    required int32 power = 7;
}

message GameState {
    required int32 game_id = 1;
    repeated Player players = 2;
}

message Island {
    required int32 id = 1;
    required IslandType island_type = 2;
    required float x = 3;
    required float y = 4;
    required int32 max_troops = 5;
    required float radius = 6;
}

message Force {
    required int32 id = 1;
    required ForceType force_type = 2;
    required float x = 3;
    required float y = 4;
    required int32 src_island = 5;
    required int32 dst_island = 6;
    required int32 uid = 7;
    required float troops = 8;
    required float morale = 9;
    required ForceMoveType move_type = 10;
    required int64 create_time = 11;
    required float radius = 12;
    required float speed = 13;
}


message GameWorld {
    required int32 width = 1;
    required int32 height = 2;
    repeated Island islands = 3;    
    repeated Force forces = 4;
}

